src/cascadia/TerminalApp/AppKeyBindingsSerialization.cpp

```diff
@@ -441,11 +441,11 @@ void winrt::TerminalApp::implementation::AppKeyBindings::LayerJson(const Json::V
 
         if (keys)
         {
-            if (!keys.isArray() || keys.size() != 1)
+            if ((!keys.isString() && !keys.isArray()) || keys.size() != 1)

```

Comment:
So when `keys` is an array with 0 elements, what happens?

Reply:
This doesn't crash.

Strange enough `{ "command": "find", "keys": [] }` is treated as an array where `keys.size() == 1`. I'm a bit confused about what `keys.size()` does but at least we don't get a crash.

Reply:
I highly disprefer relying on an explanation that amounts to "this works for some weird reason today, why exactly that is we don't really know".
Can you investigate?

Reply:
things that work by heuristic and chance have a way of spontaneously stopping working when you least expect

Reply:
Ok. I had a bad read on the debugger. It's still handled like before...
- `isString()`: false
- `isArray()`: true
- `size()`: 0
`(!keys.isArray() || keys.size() != 1)` --> `true` --> continue and skip this entry
`((!keys.isString() && !keys.isArray()) || keys.size() != 1)` --> `true` --> continue and skip this entry

Additionally, in the case where we return an empty string, `isString()` is true, `isArray()` is false, but `size()` remains as 0, resulting in the if-statements above returning `true`, etc...

Reply:
I'm blocking for this. The string size() can't be restricted to just `1` :smile: